pkg/agent/controller/networkpolicy/reject.go

```diff
@@ -0,0 +1,250 @@
+
+	ICMPv6DstUnreachableType     uint8 = 1
+	ICMPv6DstAdminProhibitedCode uint8 = 1
+)
+
+type RejectType int
+
+const (
+	// RejectPodLocal represents this packetOut is used to reject Pod-to-Pod traffic
+	// and for this response, the srcPod and the dstPod are on the same Node.
+	RejectPodLocal RejectType = iota
+	// RejectPodRemoteToLocal represents this packetOut is used to reject Pod-to-Pod
+	// traffic and for this response, the srcPod is on a remote Node and the dstPod is
+	// on the local Node.
+	RejectPodRemoteToLocal
+	// RejectLocalToRemote represents this packetOut is used to reject traffic and for
+	// this response, the srcPod is on the local Node and the dstPod is on a remote Node.
+	// While generating rejection from local to remote, there is no difference between
+	// Service traffic and Pod traffic.
+	RejectLocalToRemote
+	// RejectServiceLocal represents this packetOut is used to reject Service traffic,
+	// when AntreaProxy is enabled. The EndpointPod and the dstPod of the reject
+	// response are on the same Node.
+	RejectServiceLocal
+	// RejectServiceRemoteToLocal represents this packetOut is used to reject Service
+	// traffic, when AntreaProxy is enabled. The EndpointPod is on a remote Node and
+	// the dstPod of the reject response is on the local Node.
+	RejectServiceRemoteToLocal
+	// RejectNoAPServiceLocal represents this packetOut is used to reject Service
+	// traffic, when AntreaProxy is disabled. The EndpointPod and the dstPod of the
+	// reject response are on the same Node.
+	RejectNoAPServiceLocal
+	// RejectNoAPServiceRemoteToLocal represents this packetOut is used to reject
+	// Service traffic, when AntreaProxy is disabled. The EndpointPod is on a remote
+	// Node and the dstPod of the reject response is on the local Node.
+	RejectNoAPServiceRemoteToLocal
+)
+
+// rejectRequest sends reject response to the requesting client, based on the
+// packet-in message.
+func (c *Controller) rejectRequest(pktIn *ofctrl.PacketIn) error {
+	// Get ethernet data.
+	srcMAC := pktIn.Data.HWDst.String()
+	dstMAC := pktIn.Data.HWSrc.String()
+
+	var (
+		srcIP  string
+		dstIP  string
+		prot   uint8
+		isIPv6 bool
+	)
+	switch ipPkt := pktIn.Data.Data.(type) {
+	case *protocol.IPv4:
+		// Get IP data.
+		srcIP = ipPkt.NWDst.String()
+		dstIP = ipPkt.NWSrc.String()
+		prot = ipPkt.Protocol
+		isIPv6 = false
+	case *protocol.IPv6:
+		// Get IP data.
+		srcIP = ipPkt.NWDst.String()
+		dstIP = ipPkt.NWSrc.String()
+		prot = ipPkt.NextHeader
+		isIPv6 = true
+	}
+
+	sIface, srcFound := c.ifaceStore.GetInterfaceByIP(srcIP)
+	dIface, dstFound := c.ifaceStore.GetInterfaceByIP(dstIP)
+	// There are two situations in which it can be determined that this is a service
+	// traffic:
+	// 1. When AntreaProxy is enabled, EpSelectedRegMark is set in ServiceEPStateField.
+	// 2. When AntreaProxy is disabled, dstIP of reject response is on the local Node
+	//    and dstMAC of reject response is antrea-gw's MAC.
```

Comment:
I think this can still use some clarification. Unless I am mistaken this is accurate:
> When AntreaProxy is disabled, dstIP of reject response is on the local Node and dstMAC of reject response is antrea-gw's MAC. In this case, the reject response is being generated for locally-originated traffic that went through kube-proxy and was re-injected into the bridge through antrea-gw.

Reply:
Thanks! Changed.